test/rpc_test.py

```diff
@@ -533,6 +533,28 @@ def test_py_user_defined(self):
         )
         self.assertEqual(ret, my_function(n, n + 1, n + 2))
 
+    def _profiler_test_with_rpc(self, use_async=False):
+        dst = (self.rank + 1) % self.world_size
+        # only run profiler on rank 0.
+        if self.rank in {0}:
+            with torch.autograd.profiler.profile() as prof:
+                if not use_async:
+                    rpc.rpc_sync("worker{}".format(dst), my_sleep_func, args=(1, ))
+                else:
+                    fut = rpc.rpc_async("worker{}".format(dst), my_sleep_func, args=(1, ))
+                    time.sleep(3)
+                    fut.wait()
+            # for both sync and async, should show ~1 second time for the RPCs.
+            print(prof.key_averages())
```

Comment:
The approx processing time might be tricky, since we don't have any guarantees on how long an RPC would take in some test environments. If we impose a very generous limit I feel that would sort of defeat the purpose of the test.